lexibank_backstromnorthernpakistan.py

```diff
@@ -1,30 +1,108 @@
     dir = Path(__file__).parent
-    id = "backstromnorthernpakistan"
+    concept_class = Concept
+    language_class = Language
 
     def cmd_download(self, **kw):
-        """
-        Download files to the raw/ directory. You can use helpers methods of `self.raw`, e.g.
-
-        >>> self.raw.download(url, fname)
-        """
         pass
 
     def cmd_install(self, **kw):
-        """
-        Convert the raw data to a CLDF dataset.
+        # read raw lexical data
+        data = []
+        for i in ["A.tsv", "B.tsv", "C.tsv", "D.tsv", "E.tsv", "F.tsv"]:
+            with open(self.dir.joinpath("raw", i).as_posix(), encoding='utf-8') as csvfile:
+                reader = csv.DictReader(csvfile, delimiter="\t")
+                raw_lexemes = [row for row in reader]
+            data.extend(raw_lexemes)
 
-        Use the methods of `pylexibank.cldf.Dataset` after instantiating one as context:
-
-        >>> with self.cldf as ds:
-        ...     ds.add_sources(...)
-        ...     ds.add_language(...)
-        ...     ds.add_concept(...)
-        ...     ds.add_lexemes(...)
-        """
+        # add information to the dataset
         with self.cldf as ds:
-            pass
+            # add sources
+            ds.add_sources(*self.raw.read_bib())
+
+            # add languages to dataset
+            lang_map = {}
+            for language in self.languages:
+                lid = slug(language['Source_ID'])
+                ds.add_language(
+                    ID=lid,
+                    Name=language['Name'],
+                    Glottolog_Name=language['Glottolog_Name'],
+                    Glottocode=language['Glottocode'],
+                    ISO639P3code=language['ISO639P3code'],
+                    Latitude=language['Latitude'],
+                    Longitude=language['Longitude'],
+                    Family=language['Family'],
+                    Location=language['Location'],
+                    Source_ID=language['Source_ID'],
+                    List_ID=language['List_ID'],
+                    List_Name=language['List_Name'],
+                    Note=language['Note']
+                )
+                lang_map[language['Source_ID'].strip()] = lid
+
+            # add concepts
+            for concept in self.concepts:
+                ds.add_concept(
+                    ID=concept['ID'],
+                    Name=concept['ENGLISH'],
+                    Concepticon_ID=concept['CONCEPTICON_ID'],
+                    Concepticon_Gloss=concept['CONCEPTICON_GLOSS'],
+                )
+
+            # add lexemes
+            for idx, entry in tqdm(enumerate(data), desc='make-cldf'):
+                entry.pop('ENGLISH')
+                entry.pop('LIST ID')
+                glossid = entry.pop('GLOSS ID')
+
+                for lang, value in entry.items():
+                    lang = lang.strip()
+
+                    for form in split_text(value, separators='/', strip=True):
```

Comment:
This is something we handle differently now, but it's better, if @chrzyki explains this, as this is quicker in person than via github.

Reply:
Will push the changes tomorrow (still need to discuss this issue with @chrzyki)! Other than that, most of the corrections have been taken care of.